src/ui_basic/panel.h

```diff
@@ -436,6 +436,9 @@ class Panel {
 	virtual SaveType save_type() const {
 		return SaveType::kNone;
 	}
+	SaveType get_save_type() const {
```

Comment:
The 2 functions probably need some documentation, and `get_save_type()` should get a more descriptive name, something like `safe_save_type()` or `current_save_type()`.

Otherwise it's tested and fixed the crash.